app/models/manageiq/providers/google/inventory/parser/cloud_manager.rb

```diff
@@ -0,0 +1,335 @@
+      :cpus        => flavor.guest_cpus,
+      :description => flavor.description,
+      :ems_ref     => flavor.name,
+      :enabled     => !flavor.deprecated,
+      :memory      => flavor.memory_mb * 1.megabyte,
+      :name        => flavor.name
+    )
+  end
+
+  def cloud_volumes
+    collector.cloud_volumes.each do |cloud_volume|
+      zone_id = parse_uid_from_url(cloud_volume.zone)
+
+      persister.cloud_volumes.build(
+        :availability_zone => persister.availability_zones.lazy_find(zone_id),
+        :base_snapshot     => persister.cloud_volume_snapshots.lazy_find(cloud_volume.source_snapshot),
+        :creation_time     => cloud_volume.creation_timestamp,
+        :description       => cloud_volume.description,
+        :ems_ref           => cloud_volume.id.to_s,
+        :name              => cloud_volume.name,
+        :size              => cloud_volume.size_gb.to_i.gigabyte,
+        :status            => cloud_volume.status,
+        :volume_type       => parse_uid_from_url(cloud_volume.type)
+      )
+
+      # Take note of the source_image_id so we can expose it in parse_instance
+      @cloud_volume_url_to_source_image_id[cloud_volume.self_link] = cloud_volume.source_image_id
+      @cloud_volume_url_to_id[cloud_volume.self_link] = cloud_volume.id.to_s
+    end
+  end
+
+  def cloud_volume_snapshots
+    collector.cloud_volume_snapshots.each do |snapshot|
+      persister.cloud_volume_snapshots.build(
+        :creation_time => snapshot.creation_timestamp,
+        :description   => snapshot.description,
+        :ems_ref       => snapshot.id.to_s,
+        :name          => snapshot.name,
+        :size          => snapshot.disk_size_gb.to_i.gigabytes,
+        :status        => snapshot.status,
+        :cloud_volume  => persister.cloud_volumes.lazy_find(snapshot.source_disk)
+      )
+
+      persister_miq_template = image(snapshot)
+
+      yield persister_miq_template, snapshot
+    end
+  end
+
+  def images
+    collector.images.each do |image|
+      persister_miq_template = image(image)
+
+      yield(persister_miq_template, image)
+    end
+  end
+
+  def image(image)
+    uid = image.id.to_s
+
+    persister.miq_templates.build(
+      :deprecated         => image.kind == "compute#image" ? !image.deprecated.nil? : false,
+      :ems_ref            => uid,
+      :location           => image.self_link,
+      :name               => image.name || uid,
+      :publicly_available => true,
+      :raw_power_state    => "never",
+      :template           => true,
+      :uid_ems            => uid,
+      :vendor             => VENDOR_GOOGLE
+    )
+  end
+
+  def instances
+    collector.instances.each do |instance|
+      uid = instance.id.to_s
+
+      flavor_uid = parse_uid_from_url(instance.machine_type)
+      zone_uid = parse_uid_from_url(instance.zone)
+
+      # TODO(mslemr) if possible, change to lazy_find (now result needed immediately)
+      flavor = persister.flavors.find(flavor_uid)
```

Comment:
@Ladas how to remove this `find`? It makes another call to collector on next line, if there is not found

Reply:
It is different call to google api

Reply:
hm, I would expect we'll collect all custom flavors as a list, not 1 by 1. Lets mark this one for later, since it will need more tweaking.